src/libraries/System.Diagnostics.Process/tests/ProcessStreamReadTests.cs

```diff
@@ -143,6 +143,7 @@ async public Task TestAsyncOutputStream_CancelOutputRead()
 
                     // Wait child process close to be sure that output buffer has been flushed
                     Assert.True(p.WaitForExit(WaitInMS), "Child process didn't close");
+                    p.WaitForExit(); // wait for event handlers to complete
```

Comment:
Could this result in hanging tests? Why not just increase WaitInMs for debug builds?

Reply:
> Why not just increase WaitInMs for debug builds?

What value would you increase it to that would guarantee that the async operations are completed by the time we dispose the Process?  The only way to guarantee that is to call WaitForExit().

Reply:
That said, I just realized my change is insufficient.  If the assert fails and throws, if the Process is in a using block, we'll try to dispose it.

Reply:
(I still think we should merge it, though.  This will help minimize the impact in the case where the tests would otherwise complete successfully.)

Reply:
How about putting p.WaitForExit() in a finally block for the cases where the assertion fails?

Reply:
If the assertion fails, there's a really good chance WaitForExit would hang, too.